zetaclient/chains/solana/observer/outbound.go

```diff
@@ -350,12 +350,16 @@ func ParseGatewayInstruction(
 	case coin.CoinType_Cmd:
 		return contracts.ParseInstructionWhitelist(instruction)
 	case coin.CoinType_ERC20:
-		inst, err := contracts.ParseInstructionWithdrawSPL(instruction)
-		if err != nil {
-			return contracts.ParseInstructionExecuteSPL(instruction)
+		if inst, err := contracts.ParseInstructionWithdrawSPL(instruction); err == nil {
+			return inst, nil
 		}
-
-		return inst, err
+		if inst, err := contracts.ParseInstructionExecuteSPL(instruction); err == nil {
+			return inst, nil
+		}
+		if inst, err := contracts.ParseInstructionExecuteSPLRevert(instruction); err == nil {
+			return inst, nil
+		}
+		return nil, errors.New("failed to parse instruction")
```

Comment:
We can maybe store all the previous error and wrap all inside this  returned error (same comment for the SOL pr)